File: Palindrome.js
const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
});
   
readline.question("Digite um numero ou palavra, para saber se é palindromo: ", value => {
  
  isPalindrome(value);
  
  readline.close();
});

function isPalindrome(param){
    let original = param;
    let reverse = '';

    for(let index = original.length -1; index >= 0; index--){
      reverse += original.charAt(index);
    }
    if (original === reverse){
      console.log(original + " é um palindromo!!");
    } else {
      console.log(original + " NÂO é um palindromo!!");
    }
    
}

File: Palindrome.java
import java.util.Scanner;
public class Palindrome{
    public static void main(String[] args) {
        String original, reverse = "";
        int length;

        Scanner scan = new Scanner(System.in);

        System.out.println("Digite um numero ou palavra, para saber se é palindromo: ");
        original = scan.nextLine();
        
        length = original.length();

        for(int index = length - 1; index >= 0; index--) {
            reverse = reverse + original.charAt(index);
        }
        if (original.equals(reverse)) {
            System.out.println(original + " é um palindromo");
        } else {
            System.out.println(original + " NÂO é um palindromo");
        }
        
    }
}